openedx/core/djangoapps/public_api/courses/tests/test_course_import_export.py

```diff
@@ -0,0 +1,416 @@
+        with open(self.good_tar) as gtar:
+            args = {"name": self.good_tar, "course-data": [gtar]}
+            resp = self.client.post(self.url, args)
+        self.assertEquals(resp.status_code, 200)
+
+        course = self.store.get_course(self.course.id)
+        self.assertIsNotNone(course)
+        display_name_after_import = course.display_name
+
+        # Check that course display name have changed after import
+        self.assertNotEqual(
+            display_name_before_import,
+            display_name_after_import
+        )
+
+        # Now check that non_staff user has his same role
+        self.assertFalse(
+            CourseInstructorRole(self.course.id).has_user(nonstaff_user)
+        )
+        self.assertTrue(
+            CourseStaffRole(self.course.id).has_user(nonstaff_user)
+        )
+
+        # Now course staff user can also successfully import course
+        self.client.login(username=nonstaff_user.username, password='foo')
+        with open(self.good_tar) as gtar:
+            args = {"name": self.good_tar, "course-data": [gtar]}
+            resp = self.client.post(self.url, args)
+        self.assertEquals(resp.status_code, 200)
+
+        # Now check that non_staff user has his same role
+        self.assertFalse(
+            CourseInstructorRole(self.course.id).has_user(nonstaff_user)
+        )
+        self.assertTrue(
+            CourseStaffRole(self.course.id).has_user(nonstaff_user)
+        )
+
+    ## Unsafe tar methods #####################################################
+    # Each of these methods creates a tarfile with a single type of unsafe
+    # content.
+    def _fifo_tar(self):
+        """
+        Tar file with FIFO
+        """
+        fifop = self.unsafe_common_dir / "fifo.file"
+        fifo_tar = self.unsafe_common_dir / "fifo.tar.gz"
+        os.mkfifo(fifop)
+        with tarfile.open(fifo_tar, "w:gz") as tar:
+            tar.add(fifop)
+
+        return fifo_tar
+
+    def _symlink_tar(self):
+        """
+        Tarfile with symlink to path outside directory.
+        """
+        outsidep = self.unsafe_common_dir / "unsafe_file.txt"
+        symlinkp = self.unsafe_common_dir / "symlink.txt"
+        symlink_tar = self.unsafe_common_dir / "symlink.tar.gz"
+        outsidep.symlink(symlinkp)
+        with tarfile.open(symlink_tar, "w:gz") as tar:
+            tar.add(symlinkp)
+
+        return symlink_tar
+
+    def _outside_tar(self):
+        """
+        Tarfile with file that extracts to outside directory.
+
+        Extracting this tarfile in directory <dir> will put its contents
+        directly in <dir> (rather than <dir/tarname>).
+        """
+        outside_tar = self.unsafe_common_dir / "unsafe_file.tar.gz"
+        with tarfile.open(outside_tar, "w:gz") as tar:
+            tar.addfile(tarfile.TarInfo(str(self.content_dir / "a_file")))
+
+        return outside_tar
+
+    def _outside_tar2(self):
```

Comment:
Agreed - I combined the functions and renamed the unsafe tar creation functions.